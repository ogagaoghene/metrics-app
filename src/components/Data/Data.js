/* CONTINENTS */
const URL = 'https://disease.sh/v3/covid-19/continents';
const URLcountry = 'https://disease.sh/v3/covid-19/countries/';

const getAllContinents = async () => {
  const continents = [];
  await fetch(URL)
    .then((response) => response.json())
    .then((result) => {
      result.map((continent) => (
        continents.push({
          id: continent.continent,
          continent: continent.continent,
          cases: continent.cases,
          todayCases: continent.todayCases,
          deaths: continent.deaths,
          todayDeaths: continent.todayDeaths,
          recovered: continent.recovered,
          todayRecovered: continent.todayRecovered,
          active: continent.active,
          critical: continent.critical,
          tests: continent.tests,
          population: continent.population,
          InfoLat: continent.continentInfo.lat,
          InfoLong: continent.continentInfo.long,
          countries: continent.countries,
        })));
    });
  return continents;
};

const getApiDataCountry = async (country) => {
  const object = await fetch(URLcountry)
    .then((response) => response.json())
    .then((data) => data.filter((obj) => obj.country === country))
    .then((info) => info.map((dta) => ({
      id: country,
      flag: dta.countryInfo.flag,
      cases: dta.cases,
      deaths: dta.deaths,
      recovered: dta.recovered,
      active: dta.active,
      critical: dta.critical,
      population: dta.population,
    })));
  return object[0];
};
export { getAllContinents, getApiDataCountry };
