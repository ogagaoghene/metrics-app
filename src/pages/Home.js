import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getAllData } from '../redux/continent';
import {
  GridContainer, MapBackground, Arrow, Title, Card, SearchBar,
} from '../components/Styles/ContinentsStyle';
import Imagen from '../components/images/mapa-mundi.png';

function Home() {
  const continents = useSelector((state) => state.reduContinent);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getAllData);
  }, []);

  const [cont, setcont] = useState(false);
  const onChangeEvent = (e) => {
    const textInput = e.target.value.toLowerCase() || '';
    const newStr = textInput[0].toUpperCase().concat(textInput.slice(1, textInput.length)) || '';
    setcont(continents.filter(
      (continent) => continent.countries.includes(newStr),
    ));
  };

  return (
    <div>
      <MapBackground>
        <Title>
          Covid-19 Metrics
        </Title>
        <img src={Imagen} alt="mapa-mundi" width={400} />
      </MapBackground>
      <SearchBar type="text" name="search" id="serchBar" placeholder="Search by country here" onChange={onChangeEvent} />
      <GridContainer>
        {cont && cont.length !== 0
          ? (
            <Card key={cont[0].continent}>
              <li style={{ position: 'relative', left: '4rem' }}>
                <Link to={`/country/${cont[0].id}`}>
                  <Arrow />
                </Link>
              </li>
              <li><h2>{cont[0].continent}</h2></li>
              <li>
                <strong>Cases:</strong>
                {cont[0].cases}
              </li>
              <li>
                <strong>Deaths:</strong>
                {cont[0].deaths}
              </li>
            </Card>
          )
          : continents.map((continent) => (
            <Card key={continent.id}>
              <li style={{ position: 'relative', left: '4rem' }}>
                <Link to={`/country/${continent.id}`}>
                  <Arrow />
                </Link>
              </li>
              <li><h2>{continent.continent}</h2></li>
              <li>
                <strong>Cases:</strong>
                {continent.cases}
              </li>
              <li>
                <strong>Deaths:</strong>
                {continent.deaths}
              </li>
            </Card>
          ))}
      </GridContainer>
    </div>
  );
}

export default Home;
