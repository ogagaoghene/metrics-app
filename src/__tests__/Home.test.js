import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Home from '../pages/Home';
import store from '../redux/configureStore';
import { getAllContinents } from '../components/Data/Data';

describe('Test: Home page', () => {
  test('Should match the snapshot', () => {
    const dom = render(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    expect(dom).toMatchSnapshot();
  });

  test('Should return an array with continents', async () => {
    const continents = await getAllContinents();
    expect(continents.length).toBe(6);
  });
});
