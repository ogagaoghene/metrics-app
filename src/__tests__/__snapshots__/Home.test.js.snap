// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Home page Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-dkzDqf fInPRA"
        >
          <h1
            class="sc-hKMtZM bkbaHp"
          >
            Covid-19 Metrics
          </h1>
          <img
            alt="mapa-mundi"
            src="mapa-mundi.png"
            width="400"
          />
        </div>
        <input
          class="sc-jSMfEi jvaqI"
          id="serchBar"
          name="search"
          placeholder="Search by country here"
          type="text"
        />
        <div
          class="sc-bczRLJ bYdHwV"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-dkzDqf fInPRA"
      >
        <h1
          class="sc-hKMtZM bkbaHp"
        >
          Covid-19 Metrics
        </h1>
        <img
          alt="mapa-mundi"
          src="mapa-mundi.png"
          width="400"
        />
      </div>
      <input
        class="sc-jSMfEi jvaqI"
        id="serchBar"
        name="search"
        placeholder="Search by country here"
        type="text"
      />
      <div
        class="sc-bczRLJ bYdHwV"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
